pkg/scheduler/api/helpers/helpers.go

```diff
@@ -43,6 +43,35 @@ func Min(l, r *api.Resource) *api.Resource {
 	return res
 }
 
+// Max is used to find the max of two resource types
+func Max(l, r *api.Resource) *api.Resource {
+	res := &api.Resource{}
+
+	res.MilliCPU = math.Max(l.MilliCPU, r.MilliCPU)
+	res.Memory = math.Max(l.Memory, r.Memory)
+
+	if l.ScalarResources == nil && r.ScalarResources == nil {
+		return res
+	}
+	res.ScalarResources = map[v1.ResourceName]float64{}
+	if l.ScalarResources != nil {
+		for lName, lQuant := range l.ScalarResources {
+			if lQuant > 0 {
+				res.ScalarResources[lName] = lQuant
+			}
+		}
+	}
+	if r.ScalarResources != nil {
+		for rName, rQuant := range r.ScalarResources {
+			if rQuant > 0 {
+				maxQuant := math.Max(rQuant, res.ScalarResources[rName])
```

Comment:
This logic may result in panic. For example, l={cpu:1, memory:2, scalarResource:{gpu: 1}}, r={cpu:2, memory:1, scalarResource:{npu: 2}} . Strongly suggest to add enough UT cases for tool funcitions.

Reply:
Besides, please add `Max` function into the doc: https://github.com/volcano-sh/volcano/blob/master/docs/design/resource-comparison.md

Reply:
That's OK if using the default value `0` if there is no key in `res.ScalarResource[rName]`